cmd/oras/root/pull.go

```diff
@@ -213,20 +222,24 @@ func doPull(ctx context.Context, src oras.ReadOnlyTarget, dst oras.GraphTarget,
 					config.Annotations[ocispec.AnnotationTitle] = configPath
 				}
 			})
-			nodes = append(nodes, *config)
+			if config.Size != ocispec.DescriptorEmptyJSON.Size || config.Digest != ocispec.DescriptorEmptyJSON.Digest || config.Annotations[ocispec.AnnotationTitle] != "" {
+				nodes = append(nodes, *config)
+			}
 		}
 
 		var ret []ocispec.Descriptor
 		for _, s := range nodes {
 			if s.Annotations[ocispec.AnnotationTitle] == "" {
-				skippedLayers++
+				if s.Digest != ocispec.DescriptorEmptyJSON.Digest || s.MediaType != ocispec.DescriptorEmptyJSON.MediaType || s.Size != ocispec.DescriptorEmptyJSON.Size {
```

Comment:
Do we need to check here as the config has already been filtered out?

Reply:
No we don't, if the config is already an empty descriptor, it's not in the nodes.